connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql
+
+import org.apache.arrow.memory.RootAllocator
+
+import org.apache.spark.connect.proto
+import org.apache.spark.sql.connect.client.{ClientSparkResult, SparkConnectClient}
+import org.apache.spark.sql.connect.client.util.Cleaner
+
+class SparkSession(private val client: SparkConnectClient, private val cleaner: Cleaner)
+    extends AutoCloseable {
+
+  private[this] val allocator = new RootAllocator()
+
+  /**
+   * Executes a SQL query using Spark, returning the result as a `DataFrame`. This API eagerly
+   * runs DDL/DML commands, but not for SELECT queries.
+   *
+   * @since 2.0.0
+   */
+  def sql(query: String): Dataset = newDataset { builder =>
+    builder.setSql(proto.SQL.newBuilder().setQuery(query))
+  }
+
+  private[sql] def newDataset(f: proto.Relation.Builder => Unit): Dataset = {
+    val builder = proto.Relation.newBuilder()
+    f(builder)
+    val plan = proto.Plan.newBuilder().setRoot(builder).build()
+    new Dataset(this, plan)
+  }
+
+  private[sql] def execute(plan: proto.Plan): ClientSparkResult = {
+    val value = client.execute(plan)
+    val result = new ClientSparkResult(value, allocator)
+    cleaner.register(result)
+    result
+  }
+
+  override def close(): Unit = {
+    client.shutdown()
+    allocator.close()
+  }
+}
+
+object SparkSession {
+  def builder(): Builder = new Builder()
+
+  private lazy val cleaner = {
+    val cleaner = new Cleaner
+    cleaner.start()
+    cleaner
+  }
+
+  class Builder() {
```

Comment:
We will need a bit more configs here :)

Reply:
@vicennial Are you on this? I understand I am adding the Result. You will help with the Dataset API etc.? Or you've already finished @hvanhovell's birthday wish PR? :)

Reply:
All of the port/host/connectionstring is part of the SparkConnectClient. We can reuse the same logic here but that depends on how the user story looks like:
Do we want the user to be able to create the SparkSession directly with host/port/connection string settings or would we want them to create the client first and then feed that client into the SparkSession? Essentially do we want the client to be user-accessible?

Reply:
I'll add these to the SparkSession for now (as in, dispatch to client internally to parse etc)